subprojects/common/http-client/src/main/kotlin/com/avito/http/internal/ServiceMetricsInterceptor.kt

```diff
@@ -0,0 +1,28 @@
+package com.avito.http.internal
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import okio.IOException
+import java.net.SocketTimeoutException
+
+internal class ServiceMetricsInterceptor(private val listener: ServiceEventsListener) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response = try {
+        val response = chain.proceed(chain.request())
+
+        val latency = response.receivedResponseAtMillis - response.sentRequestAtMillis
+
+        listener.onResponse(
+            code = response.code,
+            latencyMs = latency
+        )
+        response
+    } catch (e: IOException) {
+        if (e is SocketTimeoutException) {
```

Comment:
I think that this distinction is a client's responsibility.
Otherwise, the network layer knows about the usage of different errors. We duplicate decisitons.

Reply:
agreed. moved this if to `StatsdServiceEventsListener`